cmd/util/ledger/reporters/account_reporter.go

```diff
@@ -178,7 +178,7 @@ func newAccountDataProcessor(
 				import FlowToken from 0x%s
 				access(all) fun main(account: Address): UFix64 {
 					let acct = getAccount(account)
-					let vaultRef = acct.capabilities.borrow<&FlowToken.Vault{FungibleToken.Balance}>(/public/flowTokenBalance)
+					let vaultRef = acct.capabilities.borrow<&{FungibleToken.Balance}>(/public/flowTokenBalance)
```

Comment:
Should these rather be `&FlowToken.Vault`, to assert that the borrowed object is actually a `FlowToken.Vault` like before, and not just some random other FT implementation?

Same for the other cases below

Reply:
So, the issue that I realized today is that currently, everyone has these paths linked as `FlowToken.Vault{FungibleToken.Balance}`, but since we removed restricted types, that type specification is invalid, right? I'm trying to think about what is going to be compatible after the upgrade. Is the migration plan currently that all capabilities that are currently linked as a restricted type will be changed to just be linked as the concrete type or as the interface type? I assumed that it would be the interface type, which is why I updated these to all use that version. 

But you're right that we need to make sure that the borrowed type is actually a `FlowToken.Vault`. We could just check the type or cast it to a `FlowToken.Vault` to make sure the type is correct, right? I guess it depends on the answer to my first question.

Reply:
As far as I understand @dsainati1's explanation on Discord https://discord.com/channels/613813861610684416/1162086721471647874/1202684991507795988, "all capabilities that are currently linked as a restricted type will be changed to just be linked as the concrete type or as the interface type" is correct, but also that necessary entitlements will be added.

This question keeps on coming up and is important. @dsainati1 is this correct? do we have this documented somewhere?

Reply:
> all capabilities that are currently linked as a restricted type will be changed to just be linked as the concrete type or as the interface type

Which one is it? The concrete type or the interface type?

Reply:
See the example in the Discord chat: 

> Only restricted types that use the deprecated form are migrated to base types; so a `Capability<&Collection{CollectionPublic}>` would become a `Capability<&Collection>`, but a `Capability<&{CollectionPublic}>` would be left as-is

According to that, `&FlowToken.Vault{FungibleToken.Balance}>` will become `&FlowToken.Vault`

Reply:
I want to bring this up in the implementation sync today as something that we will probably want to have a migration for. Right now, most accounts have their vaults/collections linked as a restricted type, like `FlowToken.Vault{FungibleToken.Receiver}` or `TopShot.Collection{NonFungibleToken.Receiver, NonFungibleToken.CollectionPublic, ...}` so with the way this migration is planned, all of these will be changed to the concrete reference type.

I would say it is more important to standardize making it possible to borrow these as just the interface type, then if you need to verify that the concrete type is the one you are expecting, you can either cast it as that type or use `getType`. If we convert all of these links to just the concrete type, borrowing them as the interface type will be impossible, and there is a lot of code out there that currently expects to borrow them as just interface types, such as most transfer transactions.